packages/testcontainers/src/containers/AppContainer/index.ts

```diff
@@ -0,0 +1,95 @@
+import { fetch, Response } from 'cross-fetch'
+
+import { DockerContainer } from '../DockerContainer'
+import { MasterNodeRegTestContainer } from '../RegTestContainer/Masternode'
+
+/**
+ * App Container
+ *
+ * Typically used for "sanity tests" - tests that are run seperately from the
+ * established unit tests to test against a simulated production environment,
+ * ensuring our containerised apps are resolving incoming requests as expected.
+ *
+ * We introduce the container environment so that they can be run via unit tests
+ * to create uniformity in our approach to testing. Reducing the cognitive
+ * complexity of having to run multiple automated solutions to ensure release quality.
+ *
+ * This solution works by building a new docker image using our root Dockerfile
+ * and running instances on random ports to support parallel testing allowing for
+ * vertical scaling in our test environments.
+ */
+export abstract class AppContainer extends DockerContainer {
+  public readonly name = this.generateName()
+
+  constructor (
+    public readonly app: string,
+    public readonly port = getRandomPort(3000, 5000),
+    public readonly blockchain: MasterNodeRegTestContainer = new MasterNodeRegTestContainer()
+  ) {
+    super(`${app}:sanity`)
+  }
+
+  public abstract start (): Promise<void>
+
+  /**
+   * Start the blockchain master node and return its hosting details to be used
+   * by the sanity containers.
+   *
+   * @returns Host details
+   */
+  public async startMasterNode (): Promise<{
+    hostRegTestIp: string
+    hostRegTestPort: string
+  }> {
+    await this.blockchain.start()
+
+    const hostRegTestIp = 'host.docker.internal' // TODO(eli-lim): Works on linux?
+    const hostRegTestPort = await this.blockchain.getPort('19554/tcp')
+
+    return { hostRegTestIp, hostRegTestPort }
+  }
+
+  /**
+   * Shuts down and removes any running sanity instance for this session aiming to
+   * reduce the amount of lingering docker containers as a result of running tests.
+   */
+  public async stop (): Promise<void> {
+    await this.container?.stop()
+    await this.container?.remove({ v: true })
+    await this.blockchain.stop()
+  }
+
+  public generateName (): string {
+    const rand = Math.floor(Math.random() * 10000000)
```

Comment:
I think we should use `uuid` to generated this random string instead of using `Math.random` to avoid clashing container names? As per https://github.com/JellyfishSDK/jellyfish/pull/1332

Reply:
Not a blocker. I have approved